src/common/transformations/src/transformations/common_optimizations/reverse_shape_and_type_infer.cpp

```diff
@@ -111,6 +137,95 @@ bool ov::pass::ReverseShapeAndTypeInfer::run_on_model(const std::shared_ptr<ov::
                 }
             }
             is_changed |= inherit_output_type(op, {0, 1});
+        } else if (std::dynamic_pointer_cast<Concat>(op)) {
+            std::vector<size_t> inp_idxs(op->get_input_size());
+            std::iota(inp_idxs.begin(), inp_idxs.end(), 0);
+            is_changed |= inherit_output_rank(op, inp_idxs);
```

Comment:
to be more precise, you can also deduce dimension sizes except one along which inputs are concatenated. Mya be to add todo?